Filament v4 compatibility

daikazu opened the pull request:
### Updated for Filament v4 Compatibility

- Updated minimum PHP version to 8.2
- Updated Filament dependency to require v4.0+
- Updated development dependencies for Laravel 11/Filament v4 compatibility
- Updated README badge to reflect Filament v4 support
- Confirmed all component classes work with Filament v4 APIs
- Added Button instantiation tests
- All existing functionality remains unchanged

### Breaking Changes

- Dropped support for Filament v3
- Minimum PHP version is now 8.2

daikazu committed:
test: remove unused example test

daikazu committed:
test: add unit tests for ButtonGroup and ToggleButton

Introduced tests to verify functionality of ButtonGroup and ToggleButton components. Ensures proper instantiation and behavior,

daikazu committed:
docs: update Filament version badge in README

Updated the Filament version badge link and label in the README to reflect the latest 4.x version.

daikazu committed:
feat: update dependencies for PHP 8.2 and Filament 4.x

Upgraded PHP requirement to 8.2 and Filament dependency to 4.0. Dev dependencies were also updated to align with the latest versions, ensuring compatibility and access to new features.

mohamedsabil83 commented:
Reminder @andrewdwallo

andrewdwallo commented:
@daikazu Thank you for the PR! Do you mind also updating the README "Registering Styles" section to now show them to register it using this in the theme.css file:
```
@source '../../../../vendor/andrewdwallo/filament-selectify/resources/views';
```

daikazu committed:
docs: update README with new theme integration instructions

Revised instructions for processing component styles, replacing Tailwind content array configuration with Filament custom theme setup guidance.

daikazu commented:
I updated the docs.

andrewdwallo commented:
> I updated the docs.

Cool thanks!
